source/js/components/search-form.js

```diff
@@ -0,0 +1,101 @@
+import fetchUtils from "@degjs/fetch-utils";
+
+const searchForm = () => {
+
+
+    const formEl = document.querySelector('.js-search-form');
+    const layoutEl = document.querySelector('.js-general-content')
+
+    const url = formEl.dataset.api;
+
+    function init() {
+        formEl.addEventListener('submit', e => {
+            e.preventDefault();
+            getData();
+        });
+    }
+
+
+    function getData() {
```

Comment:
[suggestion] It's not functionally wrong, but usability-wise, consider locking the form down and indicating that an async operation is in progress. You will also need to un-block it after success/error.

Reply:
You got it! added!